umbraco/Umbraco-CMS
# Improve GetPagedDescendants() query performance with an overload that does not do a count on each fetch

### Prerequisites


If there's an existing issue for this PR then this fixes 

### Description
What?
Add  GetPagedDescendants overload to media/document/repositories that does not do a count. 
Replace the existing GetPagedDescendants() where the count is not used.

Why?
Existing calls to GetPagedDescendants() are inefficient. Every time you fetch a page you run a count(*) on the entire result set. This count(*) is also inefficient as it selects on unnecessary columns and may join unnecessary tables.

How to test?
Existing tests pass.

```
/// <summary>
        /// Gets a collection of <see cref="IMedia"/> objects by Parent Id
        /// </summary>
        /// <param name="id">Id of the Parent to retrieve Descendants from</param>
        /// <param name="pageIndex">Page number</param>
        /// <param name="pageSize">Page size</param>
        /// <param name="totalRecords">Total records query would return without paging</param>
        /// <param name="ordering"></param>
        /// <param name="filter"></param>
        /// <returns>An Enumerable list of <see cref="IContent"/> objects</returns>
        IEnumerable<IMedia> GetPagedDescendants(int id, long pageIndex, int pageSize,
            IQuery<IMedia> filter = null, Ordering ordering = null);
    }
```
